test/models/network.js

```diff
@@ -147,4 +147,130 @@ describe("Network", function() {
 			);
 		});
 	});
+
+	describe("#addChannel(newChan)", function() {
+		it("should add channel", function() {
+			const chan = new Chan({name: "#thelounge"});
+
+			const network = new Network({
+				channels: [
+					chan,
+				],
+			});
+			// Lobby and initial channel
+			expect(network.channels.length).to.equal(2);
+
+			const newChan = new Chan({name: "#freenode"});
+			network.addChannel(newChan);
+
+			expect(network.channels.length).to.equal(3);
+		});
+
+		it("should add channel alphabetically", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#thelounge"});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+				],
+			});
+
+			const newChan = new Chan({name: "#freenode"});
+			network.addChannel(newChan);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(newChan);
+			expect(network.channels[3]).to.equal(chan2);
+		});
+
+		it("should sort case-insensitively", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+				],
+			});
+
+			const newChan = new Chan({name: "#freenode"});
+			network.addChannel(newChan);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(newChan);
+			expect(network.channels[3]).to.equal(chan2);
+		});
+
+		it("should sort users separately from channels", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+
+			const network = new Network({
+				channels: [
+					chan1,
+					chan2,
+				],
+			});
+
+			const newUser = new Chan({name: "mcinkay"});
+			network.addChannel(newUser);
+
+			expect(network.channels[1]).to.equal(chan1);
+			expect(network.channels[2]).to.equal(chan2);
+			expect(network.channels[3]).to.equal(newUser);
+		});
+
+		it("should sort users alphabetically", function() {
+			const chan1 = new Chan({name: "#abc"});
+			const chan2 = new Chan({name: "#THELOUNGE"});
+			const user1 = new Chan({name: "astorije"});
```

Comment:
You didn't set query type. While it's not important in this test, include it for correctness.

Reply:
Ah yes, will add that.